src/memgraph.cpp

```diff
@@ -533,16 +533,23 @@ class BoltSession final : public memgraph::communication::bolt::Session<memgraph
   using memgraph::communication::bolt::Session<memgraph::communication::v2::InputStream,
                                                memgraph::communication::v2::OutputStream>::TEncoder;
 
-  void BeginTransaction() override { interpreter_.BeginTransaction(); }
+  void BeginTransaction(const std::map<std::string, memgraph::communication::bolt::Value> &metadata) override {
+    std::map<std::string, memgraph::storage::PropertyValue> metadata_pv;
+    for (const auto &kv : metadata) metadata_pv.emplace(kv.first, memgraph::glue::ToPropertyValue(kv.second));
```

Comment:
1. Did you consider structured bindings instead of `kv`?
2. This is just nitpicking, but please use the curly brackets with loops, even if the compound statement is not required.